core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -485,6 +486,19 @@ abstract class AppBase(
     processEventsInternal()
     postCompletion()
   }
+
+  /**
+   * Sets the spark runtime based on the properties of the application.
+   */
+  private def setSparkRuntime(): Unit = {
```

Comment:
nit: is it possible to make this function of the enum object?

Reply:
Move it to enum object. I also refactored to  use `getSparkRuntime` instead of the variable `sparkRuntime`. This is because it is derived from other cached properties.